pkg/sampler/cache.go

```diff
@@ -162,6 +171,54 @@ func (s *sampleCache) getAllSamples() (allSampleInstants, int) {
 	return all, count
 }
 
+func ContainerKeyFromContainerInfo(info v1.ContainerInfo) ContainerKey {
+	labels := info.Spec.Labels
+	containerName := labels[cadvisor.ContainerLabelContainerName]
+	podNamespace := labels[cadvisor.ContainerLabelPodNamespace]
+	podName := labels[cadvisor.ContainerLabelPodName]
+	podUID := labels[cadvisor.ContainerLabelPodUID]
+	return ContainerKey{
+		ContainerID:   containerName,
+		NamespaceName: podNamespace,
+		PodName:       podName,
+		PodUID:        podUID,
+		ContainerName: containerName,
+	}
+}
+
+func (s *sampleCache) fetchCAdvisorSample(samples *sampleInstants) error {
+	if samples == nil {
+		return nil
+	}
+	containers, err := s.CadvisorClient.SubcontainersInfo("kubelet", &v1.ContainerInfoRequest{})
+	if err != nil {
+		return fmt.Errorf("failed to retrieve kubelet containers: %w", err)
+	}
+
+	for _, container := range containers {
+		// Assert pod containers.
+		key := ContainerKeyFromContainerInfo(container)
+		if key.PodUID == "" {
+			continue // Skip all non-pod containers.
+		}
+
+		// Assert containers for current samples.
+		sample, found := samples.containers[key]
+		if !found {
+			// Skip containers that are not found in the provided samples.
```

Comment:
Might be worth logging these if it's something we don't expect to see

Reply:
Done.